flink/src/main/java/org/apache/iceberg/flink/FlinkCatalog.java

```diff
@@ -80,14 +85,16 @@ public FlinkCatalog(
       String catalogName,
       String defaultDatabase,
       String[] baseNamespace,
-      Catalog icebergCatalog,
+      CatalogLoader catalogLoader,
       boolean cacheEnabled) {
     super(catalogName, defaultDatabase);
-    this.originalCatalog = icebergCatalog;
-    this.icebergCatalog = cacheEnabled ? CachingCatalog.wrap(icebergCatalog) : icebergCatalog;
+    this.originalCatalog = catalogLoader.loadCatalog(
+        HadoopUtils.getHadoopConfiguration(GlobalConfiguration.loadConfiguration()));
```

Comment:
How about  adding a `Configuration` argument in this current constructor.  IMO  for `FlinkCatalog`,  it shouldn't handle the logic about configuration initialization.  Moving the configuration loading to the `FlinkCatalogFactory`  sounds more reasonable.   besides,   unit test for `FlinkCatalog`  will be easy because it don't depend on how the flink will load its configuration.

Reply:
We can discuss it in https://github.com/apache/iceberg/pull/1332